### PR TITLE
Flush series

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This is the third part of reducing the memory used by flushSeries.
For general context see the description of #10046 (part1) and #10063 (part2).

When starting the flush of series, the Agent is now able to serialize the series already flushed without having to wait that all series are flushed. The maximum number of series in memory is reduced.
 
Only V1 series are handled.

### Motivation

Reduce memory usage of the Agent when flushing.

### Additional Notes

**  **Reviewing commit by commit is strongly recommended.** **

### Describe how to test/QA your changes

Write a DogStatsD application that send metrics (gauge, counter) during 60 seconds and check the values are correct in the backend:
- Test1: For Agent `7.33.0`
- Test2: When setting `aggregator_flush_metrics_and_serialize_in_parallel` at `false`
- Test3: When setting `aggregator_flush_metrics_and_serialize_in_parallel` at `true`
In all cases the 3 cases, results should be the same. 
In addition, the telemetries reported by `addFlushCount("Series", int64(count))` and `updateSerieTelemetry` should be the same across the 3 tests.

Check `log_payloads` still works when `aggregator_flush_metrics_and_serialize_in_parallel` is set at `true`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/aggregator/aggregator.go [hunk 1]
```diff
@@ -593,6 +601,27 @@ func (agg *BufferedAggregator) sendSeries(start time.Time, series metrics.Series
 	}
 }
 
+type iterableSerieInfo struct {
```
Comment (ogaca-dd) [inline pkg/aggregator/aggregator.go:604]: Note: `iterableSerieInfo` is removed in the next commit.

**Review by vickenty — APPROVED** [review]
Impressive! Just a few nits from me.

File: pkg/metrics/iterable_series.go [hunk 2]
```diff
@@ -89,3 +97,8 @@ func (series *IterableSeries) MoveNext() bool {
 	series.current, ok = <-series.c
 	return ok
 }
+
+// Current returns the current serie.
+func (series *IterableSeries) Current() *Serie {
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/metrics/iterable_series.go:102]: Indeed, but I prefer to keep it as I am going to use it in the next PR.

File: pkg/metrics/iterable_series.go [hunk 3]
```diff
@@ -43,6 +46,11 @@ func (series *IterableSeries) Append(serie *Serie) {
 	}
 }
 
+// GetSeriesAppenedCount returns the number of series appened with `IterableSeries.Append`.
+func (series *IterableSeries) GetSeriesAppenedCount() uint64 {
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/metrics/iterable_series.go]: Updated.

File: pkg/metrics/series.go [hunk 4]
```diff
@@ -527,12 +527,12 @@ func (series Series) WriteHeader(stream *jsoniter.Stream) error {
...
 }
 
-// WriteFooter writes the payload footer for this type
+// WriteFooter prints the payload footer for this type
 func (series Series) WriteFooter(stream *jsoniter.Stream) error {
 	return writeFooter(stream)
 }
 
-// WriteItem writes the json representation of an item
+// WriteItem prints the json representation of an item
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/metrics/series.go]: Updated.

File: pkg/config/config.go [hunk 5]
```diff
@@ -367,6 +367,9 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("aggregator_stop_timeout", 2)
 	config.BindEnvAndSetDefault("aggregator_buffer_size", 100)
 	config.BindEnvAndSetDefault("basic_telemetry_add_container_tags", false) // configure adding the agent container tags to the basic agent telemetry metrics (e.g. `datadog.agent.running`)
+	config.BindEnvAndSetDefault("flush_metrics_and_serialize_in_parallel", false)
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/config/config.go]: Code updated.

**Review by vickenty — APPROVED** [review]

**Review by vickenty — APPROVED** [review]
LGTM

**Pull request merged**